superset-frontend/src/components/Badge/index.tsx

```diff
@@ -28,14 +28,25 @@ const Badge = styled(
   (
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     { textColor, color, text, ...props }: BadgeProps,
-  ) => <AntdBadge text={text} color={text ? color : undefined} {...props} />,
+  ) => (
+    <AntdBadge
+      text={text}
+      color={text ? color : undefined}
+      status={props.status || 'default'}
+      {...props}
+    />
+  ),
 )`
-  & > sup {
-    padding: 0 ${({ theme }) => theme.gridUnit * 2}px;
-    background: ${({ theme, color }) => color || theme.colors.primary.base};
-    color: ${({ theme, textColor }) =>
-      textColor || theme.colors.grayscale.light5};
-  }
+  ${({ theme, color, count }) => `
+    & > sup,
+    & > sup.ant-badge-count {
+      ${
+        count !== undefined
+          ? `background: ${color || theme.colors.primary.base};`
```

Comment:
This is an example of a necessary customization for backward compatibility. Ant Design uses `colorError` as the background color for the count component of the Badge. However, in Superset, the count requires a different color. Changing the `colorError` of the Badge in the Ant Design `ConfigProvider` would affect all variations of the Badge, which is not desirable. See all variations here: https://ant.design/components/badge.